code/parachain/runtime/primitives/src/currency.rs

```diff
@@ -89,9 +89,9 @@ macro_rules! list_assets {
 			}
 		}
 
-		pub fn list_assets() -> Vec<Asset> {
+		pub fn list_assets() -> Vec<Asset<XcmAssetLocation>> {
 			[
-				$(Asset { id: CurrencyId::$NAME.0 as u64, name: stringify!($NAME).as_bytes().to_vec() },)*
+				$(Asset { id: CurrencyId::$NAME.0 as u128, name: Some(stringify!($NAME).as_bytes().to_vec()), decimals: 12_u8.into(), foreign_id: None },)*
```

Comment:
KSM and USD* have foreign asset id too or will have as soon as registered.

Reply:
so it  there should be join of registry with hard coded

Reply:
@dzmitry-lahoda  What can we use to identify already existing assets, given that when we register we are not giving any input on `local_id` or `name` or `symbol`?

Reply:
Encoding works like that https://drive.google.com/file/d/169Jb6bdJKIGtK0bVqYlQgDSvN0nSfNhq/view  and can see some examples https://github.com/ComposableFi/composable/pull/2035#issuecomment-1288758296

Reply:
will update after check

Reply:
Thanks @dzmitry-lahoda , it helped me understanding better xcm.
Can you provide me with a concrete example on how a `MultiLocation` would look like when we register on our chain for example `USDT` (which locally has id `130`) , which let's assume has id `111` on some parachain with id `222`?